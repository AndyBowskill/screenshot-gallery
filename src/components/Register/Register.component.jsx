import React from 'react';
import './Register.style.css';

class Register extends React.Component {
  constructor() {
    super();
    this.state = {
      email: '',
      name: '',
      password: '',
    };
  }

  onEmailChange = (event) => {
    this.setState({ email: event.target.value });
  };

  onNameChange = (event) => {
    this.setState({ name: event.target.value });
  };

  onPasswordChange = (event) => {
    this.setState({ password: event.target.value });
  };

  onRegisterSumbit = () => {
    fetch('https://screenshot-gallery-api.herokuapp.com/register', {
      method: 'post',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        email: this.state.email,
        name: this.state.name,
        password: this.state.password,
      }),
    })
      .then((response) => response.json())
      .then((user) => {
        if (user.id) {
          this.props.loadUser(user);
          this.props.loadScreenshots([]);
          this.props.onRouteChange('home');
        }
      });
  };

  render() {
    return (
      <section className='register flex'>
        <div>
          <h3>Register</h3>

          <label htmlFor='email'>Email</label>
          <input
            type='text'
            placeholder='Please enter email'
            name='email'
            id='email'
            required
            onChange={this.onEmailChange}
          />

          <label htmlFor='name'>Name</label>
          <input
            type='text'
            placeholder='Please enter name'
            name='name'
            id='name'
            required
            onChange={this.onNameChange}
          />

          <label htmlFor='psw'>Password</label>
          <input
            type='password'
            placeholder='Please enter password'
            name='psw'
            id='psw'
            required
            onChange={this.onPasswordChange}
          />

          <button type='submit' onClick={this.onRegisterSumbit}>
            Register
          </button>
        </div>
      </section>
    );
  }
}

export default Register;
