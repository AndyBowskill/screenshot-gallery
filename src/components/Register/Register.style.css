.register.flex {
  display: flex;
  justify-content: center;
  padding: 16px;
}

.register.flex h3 {
  font-size: 1rem;
}

.register input {
  font-family: 'Didact Gothic', sans-serif;
  font-size: 1rem;
  border: 1px solid #0074d9;
  border-radius: 5px;
  width: 340px;
  display: block;
  box-sizing: border-box;
  padding: 16px;
  margin: 12px 0 12px 0;
}

.register button {
  font-family: 'Didact Gothic', sans-serif;
  font-size: 1rem;
  background-color: rgb(127, 219, 255);
  color: #0074d9;
  padding: 12px 30px;
  margin: 12px 0 24px 0;
  text-align: center;
  text-decoration: none;
  border: 1px solid #0074d9;
  cursor: pointer;
  border-radius: 5px;
  width: 340px;
  display: block;
  opacity: 0.9;
}

.register button:hover {
  opacity: 1;
}

@media only screen and (min-width: 480px) {
  .register.flex h3 {
    font-size: 1.2rem;
  }
}
