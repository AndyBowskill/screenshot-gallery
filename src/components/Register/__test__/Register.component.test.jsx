import React from 'react';
import renderer from 'react-test-renderer';
import Register from '../Register.component';
import { render, screen, fireEvent } from '@testing-library/react';

const onRouteChange = jest.fn();
const loadUser = jest.fn();
const loadScreenshots = jest.fn();

test('Register component displays correctly.', () => {
  const component = renderer.create(
    <Register
      onRouteChange={onRouteChange}
      loadUser={loadUser}
      loadScreenshots={loadScreenshots}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Email element changes correctly.', () => {
  render(
    <Register
      onRouteChange={onRouteChange}
      loadUser={loadUser}
      loadScreenshots={loadScreenshots}
    />
  );

  const emailElement = screen.getByPlaceholderText(/Please enter email/);

  fireEvent.change(emailElement, { target: { value: 'andy@gmail.com' } });
});

test('Name element changes correctly.', () => {
  render(
    <Register
      onRouteChange={onRouteChange}
      loadUser={loadUser}
      loadScreenshots={loadScreenshots}
    />
  );

  const nameElement = screen.getByPlaceholderText(/Please enter name/);

  fireEvent.change(nameElement, { target: { value: 'John Smith' } });
});

test('Password element changes correctly.', () => {
  render(
    <Register
      onRouteChange={onRouteChange}
      loadUser={loadUser}
      loadScreenshots={loadScreenshots}
    />
  );

  const passwordElement = screen.getByPlaceholderText(/Please enter password/);

  fireEvent.change(passwordElement, { target: { value: '123' } });
});

test('Register button click event is successful.', () => {
  render(
    <Register
      onRouteChange={onRouteChange}
      loadUser={loadUser}
      loadScreenshots={loadScreenshots}
    />
  );

  const registerElement = screen.getAllByText('Register');

  fireEvent.click(registerElement[1]);
});
