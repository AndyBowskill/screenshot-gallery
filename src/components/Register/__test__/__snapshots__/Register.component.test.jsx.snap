// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component displays correctly. 1`] = `
<section
  className="register flex"
>
  <div>
    <h3>
      Register
    </h3>
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Please enter email"
      required={true}
      type="text"
    />
    <label
      htmlFor="name"
    >
      Name
    </label>
    <input
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Please enter name"
      required={true}
      type="text"
    />
    <label
      htmlFor="psw"
    >
      Password
    </label>
    <input
      id="psw"
      name="psw"
      onChange={[Function]}
      placeholder="Please enter password"
      required={true}
      type="password"
    />
    <button
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
  </div>
</section>
`;
